### ISSUE TITLE
MEGA CD games freeze after loading save

### ISSUE THREAD

**Issue by THEWRCN** [irixxxx/picodrive#100]
First of all, I love this emulator and thank you for doing a great job.

So to my problem: 

I am going crazy over here, reading the previous issue here as well.  Whenever I play Mega cd games, saving the gaming is no problem, but sometimes loading the game, it can freeze, the music is still playing, but the game has frozen. This especially happens when I am playing the Lunar games, and I have to start all over again. 

Is there any solution to my problem, any workaround? Any help will be appreciated, 

I am using Miyoo CFW on Q90, Picodrive 1.93, but binary 1.99. 

Thank you all

King regards

THEWRCN

**Comment by irixxxx** [comment]
Please give a description of what I need to do to reproduce this on the miyoo.
Attach one of the damaged save files.
Which firmware are you using?

There is already another ticket for this. Could you try to verify if this is really related to the date settings?

**Comment by THEWRCN** [comment]
Thank you for your message. 

I use miyoo cfw v1.3.3. I only use savestats when I save games. On normal Mega Drive games, it is working fine. The problem occurs when I am only playing CD games, Snatcher, Lunar games etc. I save via savestate and whenever I want to play, I load the game via loadstate, the game freeze when you enter a building, etc, but the music is still running. 

I checked the another ticket, it did not work for me, that is why I created this ticket, and I will check out if it is date related issue. I do not have the damaged save file, but I will post it if game freeze again. 

Thank you

**Comment by THEWRCN** [comment]
Hey all

Yeah, I checked it, it is not related to the time settings, I changed the time every time  before I booted the game. 

I do not know when one these two are saving file, but I am attaching both of them, if they are wrong, let me know, I will attach the correct one. 
[Lunar - The Silver Star (USA).mds.gz](https://github.com/irixxxx/picodrive/files/13440703/Lunar.-.The.Silver.Star.USA.mds.gz)

Thanks

**Comment by THEWRCN** [comment]
[Lunar - The Silver Star (USA).zip](https://github.com/irixxxx/picodrive/files/13440720/Lunar.-.The.Silver.Star.USA.zip)

**Comment by irixxxx** [comment]
What exactly is happening for you when you load this savestate?

**Comment by THEWRCN** [comment]
When I load this savestate, the game will freeze if I enter a house or press the menu, but the music is still playing. If a enter a house or dungeon, the game only display black screen while music is still going on. This is regarding Lunar games. Snatcher freezes completely, and I have to start all over.

**Comment by irixxxx** [comment]
Yes, the bug occurs for me as well. I cannot reproduce creating such a save file though. To check if this is some bug in the ARM-specific parts of PicoDrive, here are 2 versions each disabling a different part of the ARM related code:

[PicoDrive-cz80.zip](https://github.com/irixxxx/picodrive/files/13466541/PicoDrive-cz80.zip)
[PicoDrive-fame.zip](https://github.com/irixxxx/picodrive/files/13466542/PicoDrive-fame.zip)

Could you please check if these also create damaged save files?

**Comment by THEWRCN** [comment]
> Yes, the bug occurs for me as well. I cannot reproduce creating such a save file though. To check if this is some bug in the ARM-specific parts of PicoDrive, here are 2 versions each disabling a different part of the ARM related code:
> 
> [PicoDrive-cz80.zip](https://github.com/irixxxx/picodrive/files/13466541/PicoDrive-cz80.zip)
> [PicoDrive-fame.zip](https://github.com/irixxxx/picodrive/files/13466542/PicoDrive-fame.zip)
> 
> Could you please check if these also create damaged save files?
> 

Thank you so much for your message, Irixxxx, but forgive my IT skills, but where should I put the files on Picodrive?

Thanks

**Comment by irixxxx** [comment]
replace the file PicoDrive on your miyoo SD card with the one in the zipfiles and conduct some testing.

FWIW, since you are using 1.99, please download the current 2.00alpha1 release file from the release page and make the same test with that as well. 1.99 is about 2 yrs behind 2.00-alpha1

**Comment by THEWRCN** [comment]
Thank you, let me check it out and I will report back. :)

**Comment by THEWRCN** [comment]
Hey, Irixxx and all

I can confirm that picodrive - fame.zip is the file that did not corrupt the savefile and is working great so far. I have saved and loaded Lunar for an hour, and no corrupt file so far. 

Thank you Irixxx for your work, now I can save in peace.

**Comment by irixxxx** [comment]
Could you conduct tests with 2.00 as well? I'd want to know if this is a bug in the ARM component I replaced in the fame version, or if the bug has been generally fixed.

**Comment by THEWRCN** [comment]
Hey Irixxxx and all. 

I have updated to 2.00.

I did put the picodrive.sh file from picodrive - fame.zip and put it in the 2.00. and tested about 10 - 15 saves and loads, and they are working great. No corrupt file, and buttons are working great as well. I will try some more, and let you know if something comes up.

Thank you again Irixxxx. Great work my man.

Regards from me.

**Comment by irixxxx** [comment]
OK, assuming this has been fixed in the past for now. Please reopen if you find something is still off.
